python/sglang/srt/two_batch_overlap.py

```diff
@@ -415,9 +468,72 @@ def prepare_raw(
             out_num_token_non_padded=out_num_token_non_padded_b,
         )
 
+        if is_enable_two_chunk:
+            cls.filter_batch_for_two_chunk(
+                batch,
+                child_a=child_a,
+                child_b=child_b,
+                tbo_split_seq_index=batch.tbo_split_seq_index,
+            )
+
         assert batch.tbo_children is None
         batch.tbo_children = [child_a, child_b]
 
+    @classmethod
+    def filter_batch_for_two_chunk(
+        cls,
+        batch: ForwardBatch,
+        *,
+        child_a: ForwardBatch,
+        child_b: ForwardBatch,
+        tbo_split_seq_index: int,
+    ):
+        extend_seq_lens_cpu = batch.extend_seq_lens_cpu
+        overall_sum = sum(extend_seq_lens_cpu)
+        half_sum = overall_sum // 2
+        left_token_num = half_sum - sum(extend_seq_lens_cpu[:tbo_split_seq_index])
+        right_token_num = extend_seq_lens_cpu[tbo_split_seq_index] - left_token_num
+
+        child_a.extend_seq_lens_cpu[-1] = left_token_num
+        child_b.extend_seq_lens_cpu[0] = right_token_num
+
+        child_a.extend_seq_lens = torch.tensor(
+            child_a.extend_seq_lens_cpu, dtype=batch.extend_seq_lens.dtype
+        ).to(device=global_server_args_dict["device"], non_blocking=True)
+        child_b.extend_seq_lens = torch.tensor(
+            child_b.extend_seq_lens_cpu, dtype=batch.extend_seq_lens.dtype
+        ).to(device=global_server_args_dict["device"], non_blocking=True)
+
+        child_a.extend_num_tokens = sum(child_a.extend_seq_lens_cpu)
+        child_b.extend_num_tokens = sum(child_b.extend_seq_lens_cpu)
```

Comment:
nit: for such derived data, maybe extract to utils

e.g. most naive is

```
def some_postprocessing(x):
  x.extend_seq_lens = x.extend_seq_lens_cpu.to
  x.extend_num_tokens = ...
```

and call it

```
some_postprocessing(child_a)
some_postprocessing(child_b)
```

or even better, is it possible we extract these derived data computation from the main code to utils, and call it (again) here